File: C/1170-compare-strings-by-frequency-of-the-smallest-character/main.go
package main

import (
	"fmt"
	"sort"
)

func numSmallerByFrequency(queries []string, words []string) []int {
	qbuf := make([]int, len(queries))
	for i, v := range queries {
		buf := make([]int, 26)
		for _, b := range v {
			buf[b-'a']++
		}
		for _, c := range buf {
			if c > 0 {
				qbuf[i] = c
				break
			}
		}
	}
	wbuf := make([]int, len(words))
	for i, v := range words {
		buf := make([]int, 26)
		for _, b := range v {
			buf[b-'a']++
		}
		for _, c := range buf {
			if c > 0 {
				wbuf[i] = c
				break
			}
		}
	}
	sort.Ints(wbuf)
	var ret []int
	for _, b := range qbuf {
		s, e := 0, len(wbuf)-1
		for s <= e {
			m := s + (e-s)/2
			if wbuf[m] <= b {
				s = m + 1
			} else if wbuf[m] > b {
				e = m - 1
			}
		}
		ret = append(ret, len(wbuf)-s)
	}
	return ret
}

func main() {
	fmt.Println("a")
}

File: C/1170-compare-strings-by-frequency-of-the-smallest-character/main.rs
struct Solution {}

impl Solution {
  pub fn num_smaller_by_frequency(queries: Vec<String>, words: Vec<String>) -> Vec<i32> {
    fn count(word: &String) -> i32 {
      let mut cnt: i32 = 0;
      let mut max: u8 = 255;
      word.as_bytes().to_vec().iter().for_each(|&b| {
        if max > b {
          max = b;
          cnt = 1;
        } else if max == b {
          cnt += 1;
        }
      });
      cnt
    }
    let mut cnt = words.iter().map(|word| count(word)).collect::<Vec<i32>>();
    cnt.sort();

    queries
      .iter()
      .map(|query| {
        let c = count(query);
        let mut start: i32 = 0;
        let mut end: i32 = cnt.len() as i32 - 1;
        while start <= end {
          let mid = start + (end - start) / 2;
          if cnt[mid as usize] > c {
            end = mid - 1;
          } else {
            start = mid + 1;
          }
        }
        cnt.len() as i32 - start
      })
      .collect::<Vec<i32>>()
  }
}
